### PR TITLE
Move to static BK pipeline for x-pack/metricbeat

### PR DESCRIPTION
**Pull request by dliappis**
## Proposed commit message

This commit refactors the x-pack/metricbeat
pipeline to the pipeline selector approach,
as laid out in #38783

## Related issues

- https://github.com/elastic/ingest-dev/issues/3072

## Logs

- no labels (mandatory tests only): https://buildkite.com/elastic/beats-xpack-metricbeat/builds/2371
- `macos` and `:Windows`: https://buildkite.com/elastic/beats-xpack-metricbeat/builds/2403
- latest commit with all labels: https://buildkite.com/elastic/beats-xpack-metricbeat/builds/2427

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -101,6 +101,32 @@ steps:
...
+  - label: "Trigger x-pack/metricbeat"
+    plugins:
+      - monorepo-diff#v1.0.1:
+          diff: "git diff --name-only origin/${GITHUB_PR_TARGET_BRANCH}...HEAD"
+          watch:
+            - path:
+                - x-pack/metricbeat/
+                - x-pack/libbeat/common/aws
+                - .buildkite/x-pack/pipeline.xpack.metricbeat.yml
+                - .buildkite/scripts
```
Comment (alexsapran) [inline .buildkite/pipeline.yml:113]: taken form the discussion today, let include also the hooks

Comment (dliappis) [reply .buildkite/pipeline.yml:113]: thanks! fixed in 7066d4cd56c0adff9402e707d2df33c3f6c1ea04

**Review by alexsapran — APPROVED** [review]
thanks

**Pull request merged**